src/sections/create-dataset/CreateDatasetContext.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { ChangeEvent, FormEvent, useState } from 'react'
+import {
+  CreateDatasetFormFields,
+  FormSubmissionService,
+  FormValidationService,
+  FormValidationResult
+} from '../../dataset/domain/useCases/createDataset'
+
+interface FormContextInterface {
+  fields: CreateDatasetFormFields
+}
+
+const defaultFormState: CreateDatasetFormFields = {
+  createDatasetTitle: ''
+}
+
+interface CreateDatasetFormProps {
+  formValidationService: FormValidationService
+  formSubmissionService: FormSubmissionService
+}
+
+export enum SubmissionStatusEnums {
+  NotSubmitted = 'NotSubmitted',
+  IsSubmitting = 'IsSubmitting',
+  SubmitComplete = 'SubmitComplete',
+  Errored = 'Errored'
+}
+
+export function useCreateDatasetForm({
+  formValidationService,
+  formSubmissionService
+}: CreateDatasetFormProps) {
+  const [formState, setFormState] = useState<FormContextInterface>({
+    fields: defaultFormState
+  })
+  const [submissionStatus, setSubmissionStatus] = React.useState<SubmissionStatusEnums>(
```

Comment:
Please use a consistent syntax. Above, you're using `useState` and below, `React.useState`. To avoid importing React, please remove the `React.` prefix and import only the necessary functions such as `useState`